src/datumaro/plugins/configurable_validator.py

```diff
@@ -0,0 +1,732 @@
+                            "Label '%s': failed to get access to attribute %s" % (label_name, e)
+                        )
+                        if {UndefinedAttribute} & self.warnings:
+                            # attribute is not defined within the label
+                            self.stats["undefined_attribute"].add(item_key + (label_name, attr))
+
+                if {MissingAttribute} & self.warnings:
+                    # defined attribute is not found in this annotation
+                    for attr in self.stats[label_name]["attributes"]:
+                        if attr not in annotation.attributes:
+                            self.stats["missing_attribute"].add(item_key + (label_name, attr))
+        else:
+            label_name = str(annotation.label)
+            if {UndefinedLabel} & self.warnings:
+                self.stats["undefined_label"].add(item_key + (label_name,))
+
+            if {UndefinedAttribute} & self.warnings:
+                for attr, value in annotation.attributes.items():
+                    self.stats["undefined_attribute"].add(item_key + (label_name, attr))
+
+    def _update_item_type_stats(self, item: DatasetItem):
+        NotImplemented
+
+    def _update_ann_type_stats(self, item_key: tuple, annotation: Annotation):
+        NotImplemented
+
+
+class ClsStats(_BaseAnnStats):
+    def __init__(self, label_categories: LabelCategories, warnings: set):
+        super().__init__(label_categories=label_categories, warnings=warnings)
+
+        self.stats["multiple_label"] = set()
+
+    def _update_item_type_stats(self, item: DatasetItem):
+        item_key = (item.id, item.subset)
+
+        if {MultiLabelAnnotations} & self.warnings:
+            if sum([1 for ann in item.annotations if ann.type == AnnotationType.label]) > 1:
+                self.stats["multiple_label"].add(item_key)
+
+    def _update_ann_type_stats(self, item_key: tuple, annotation: Annotation):
+        pass
+
+
+class DetStats(_BaseAnnStats):
+    BBOX_WARNINGS = {
+        FarFromAttrMean,  # annotation level
+        FarFromLabelMean,  # annotation level
+        ImbalancedDistInAttribute,  # annotation level: bbox
+        ImbalancedDistInLabel,  # annotation level: bbox
+        InvalidValue,  # annotation level
+        NegativeLength,  # annotation level
+    }
+
+    def __init__(self, label_categories: LabelCategories, warnings: set):
+        super().__init__(label_categories=label_categories, warnings=warnings)
+
+        self.items = {}
+        for label_cat in self.label_categories.items:
+            label_name = label_cat.name
+            self.stats[label_name]["pts"] = []  # (id, subset, ann_id, w, h)
+
+        self.stats["invalid_value"] = set()
+        self.stats["negative_length"] = set()
+
+    def _update_item_type_stats(self, item: DatasetItem):
+        pass
+
+    def _update_ann_type_stats(self, item_key: tuple, annotation: Annotation):
+        if self.BBOX_WARNINGS & self.warnings:
```

Comment:
I hope that we build a collection of update functions when initializing this class instance according to the input configuration. This is for doing not to check its configuration every time during the dataset item loop. We can iterate on the collection of update functions.

Reply:
I have tried to register function for warning when initialization at cc10ff43ab54b743fc524f29ad1f4bc016eaa04b. Could you check this?